readium/shared/src/main/java/org/readium/r2/shared/publication/protection/ContentProtection.kt

```diff
@@ -7,56 +7,65 @@
  * LICENSE file present in the project repository where this source code is maintained.
  */
 
-package org.readium.r2.shared.publication
+package org.readium.r2.shared.publication.protection
 
 import androidx.annotation.StringRes
 import org.readium.r2.shared.R
 import org.readium.r2.shared.UserException
+import org.readium.r2.shared.error.Try
 import org.readium.r2.shared.fetcher.Fetcher
-import org.readium.r2.shared.publication.asset.PublicationAsset
+import org.readium.r2.shared.publication.LocalizedString
+import org.readium.r2.shared.publication.Publication
 import org.readium.r2.shared.publication.services.ContentProtectionService
-import org.readium.r2.shared.util.Try
+import org.readium.r2.shared.util.mediatype.MediaType
 
 /**
  * Bridge between a Content Protection technology and the Readium toolkit.
  *
  * Its responsibilities are to:
- * - Unlock a publication by returning a customized [Fetcher].
+ * - Create a [Fetcher] one can access the publication through.
  * - Create a [ContentProtectionService] publication service.
  */
 interface ContentProtection {
 
+    val scheme: Scheme
+
+    /**
+     * Returns if this [ContentProtection] supports the given [asset].
+     */
+    suspend fun supports(
+        asset: org.readium.r2.shared.asset.Asset
+    ): Boolean
+
     /**
      * Attempts to unlock a potentially protected publication asset.
      *
-     * @return A [ProtectedAsset] in case of success, null if the asset is not protected by this
+     * @return A [Asset] in case of success, null if the asset is not protected by this
      * technology or a [Publication.OpeningException] if the asset can't be successfully opened,
      * even in restricted mode.
      */
     suspend fun open(
-        asset: PublicationAsset,
+        asset: org.readium.r2.shared.asset.Asset,
+        drmScheme: String,
```

Comment:
How about using `drmScheme: Scheme` instead for better type safety?

Also, I'm starting to think that we should have just `value class Scheme(val uri: String)` instead of having a localized name with it. It will be easier to get a proper instance of `Scheme` from a raw URI, and we can let the app decide how to represent the name of a DRM.

Reply:
There is  the same issue with restoring `MediaType`s from strings, though that's fine in practice. `name` is useless, but `fileExtension` is not.

Reply:
Sounds good!